server/GraknServer.java

```diff
@@ -147,52 +151,64 @@ private static Properties parseProperties() {
+                return null;
             } else {
-                proceed = true;
+                if (parseResult.hasSubcommand()) {
+                    assert parseResult.subcommand().asCommandLineList().size() == 1;
+                    return parseResult.subcommand().asCommandLineList().get(0).getCommand();
+                } else {
+                    assert parseResult.asCommandLineList().size() == 1;
+                    return parseResult.asCommandLineList().get(0).getCommand();
+                }
             }
         } catch (ParameterException ex) {
-            command.getErr().println(ex.getMessage());
-            if (!UnmatchedArgumentException.printSuggestions(ex, command.getErr())) {
-                ex.getCommandLine().usage(command.getErr());
+            commandLine.getErr().println(ex.getMessage());
+            if (!UnmatchedArgumentException.printSuggestions(ex, commandLine.getErr())) {
+                ex.getCommandLine().usage(commandLine.getErr());
             }
-            proceed = false;
+            return null;
         }
-
-        if (proceed) return options;
-        else return null;
     }
 
     public static void main(String[] args) {
         try {
             final long start = System.nanoTime();
 
             printASCIILogo();
-            final ServerOptions options = parseCommandLine(parseProperties(), args);
-            if (options == null) System.exit(0);
-
-            final GraknServer server = new GraknServer(options);
-            server.start();
-
-            final long end = System.nanoTime();
-            LOG.info("Grakn Core version: {}", Version.VERSION);
-            LOG.info("Grakn Core Server has been started (in {} ms)",
-                     String.format("%.3f", (end - start) / 1_000_000.00));
-
-            server.serve();
+            final ServerCommand command = parseCommandLine(parseProperties(), args);
+            if (command == null) System.exit(0);
+
+            if (command instanceof ServerCommand.Start) {
+                final GraknServer server = new GraknServer(command.asStart());
+                server.start();
+
+                final long end = System.nanoTime();
+                LOG.info("Grakn Core version: {}", Version.VERSION);
+                LOG.info("Grakn Core Server has been started (in {} ms)",
+                        String.format("%.3f", (end - start) / 1_000_000.00));
+
+                server.serve();
+            } else if (command instanceof ServerCommand.ImportData) {
+                ServerCommand.ImportData importDataCommand = command.asImport();
+                MigratorClient migrator = new MigratorClient(importDataCommand.port());
+                boolean success = migrator.importData(importDataCommand.database(), importDataCommand.filename(), importDataCommand.remapLabels());
+                System.exit(success ? 0 : 1);
+            } else {
+                assert false;
```

Comment:
I think you show throw an exception here, such as invalid state, or something similar rather than use an assertion

Reply:
Actually I can just delete the else clause.